### PR TITLE
Cleanup in native File classes

### PR DESCRIPTION
**Pull request by marcusnaslund**
As a part of #1662 - changing as much as possible without affecting the API. Can't believe I haven't touched these files before.

Remember to append `?w=1` to the URL when viewing the diff.

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Peer review @sebastianbaginski ?

File: source/system/native/FileUnix.ooc [hunk 1]
```diff
@@ -27,313 +30,219 @@ readdir_r: extern func (DIR*, DirEnt*, DirEnt**) -> Int
...
+		if (exec)
+			mode |= (S_IXUSR | S_IXGRP | S_IXOTH)
+		else
+			mode &= ~(S_IXUSR | S_IXGRP | S_IXOTH)
+		result := fchmod(fd, mode) == 0
+		if (close(fd) != 0)
+			result = false
+		result
+	}
+	lastAccessed: override func -> Long {
```
Comment (sebastianbaginski) [inline source/system/native/FileUnix.ooc]: Maybe we should change these functions to return `DateTime` instead ? Not this PR of course.

Comment (marcusnaslund) [inline source/system/native/FileUnix.ooc]: Good idea.

Comment (marcusnaslund) [inline source/system/native/FileUnix.ooc]: Will require us to move `DateTime` away from `base`, though.

Comment (sebastianbaginski) [inline source/system/native/FileUnix.ooc]: You decide if it's worth it, I don't think anyone uses these functions now.

Comment (marcusnaslund) [inline source/system/native/FileUnix.ooc]: I think not now. We will revisit this later if we need it.

**Pull request merged**